internal/util/bundleutil/bundleutil.go

```diff
@@ -0,0 +1,239 @@
+	defaultBundleDockerfilePath = "bundle.Dockerfile"
+)
+
+// BundleMetaData contains the required metadata to build bundles and images.
+type BundleMetaData struct {
+	// BundleDir refers to the directory where generated bundles are to be written.
+	BundleDir string
+
+	// The PackageName of the operator bundle.
+	PackageName string
+
+	// Channels and DefaultChannel the operator shoud be subscribed to.
+	Channels       string
+	DefaultChannel string
+
+	// BaseImage name to build bundle image.
+	BaseImage string
+
+	// BuildCommand to run while building image.
+	BuildCommand string
+
+	// PackageManifestPath where the input manifests are present.
+	PkgmanifestPath string
+
+	// Other labels to be added in CSV.
+	OtherLabels map[string]string
+}
+
+// values to populate bundle metadata/Dockerfile.
+type annotationsValues struct {
+	BundleDir      string
+	PackageName    string
+	Channels       string
+	DefaultChannel string
+	OtherLabels    []string
+}
+
+// GenerateMetadata scaffolds annotations.yaml and bundle.Dockerfile with the provided
+// annotation values.
+func (meta *BundleMetaData) GenerateMetadata() error {
+	// Create output directory
+	if err := os.MkdirAll(meta.BundleDir, projutil.DirMode); err != nil {
+		return err
+	}
+
+	// Create annotation values for both bundle.Dockerfile and annotations.yaml, which should
+	// hold the same set of values always.
+	values := annotationsValues{
+		BundleDir:      meta.BundleDir,
+		PackageName:    meta.PackageName,
+		Channels:       meta.Channels,
+		DefaultChannel: meta.DefaultChannel,
+	}
+
+	for k, v := range meta.OtherLabels {
+		values.OtherLabels = append(values.OtherLabels, fmt.Sprintf("%s=%s", k, v))
+	}
+
+	// Write each file
+	metadataDir := filepath.Join(meta.BundleDir, defaultMetadataDir)
+	if err := os.MkdirAll(metadataDir, projutil.DirMode); err != nil {
+		return err
+	}
+
+	dockerfilePath := defaultBundleDockerfilePath
+	// If migrating from packagemanifests to bundle, bundle.Docker file is present
+	// inside bundleDir, else its in the project directory.
+	// Remmove this, when pkgman-to-bundle migrate command is removed.
+	if len(meta.PkgmanifestPath) != 0 {
+		dockerfilePath = filepath.Join(meta.BundleDir, "bundle.Dockerfile")
+	}
+
+	templateMap := map[string]*template.Template{
+		dockerfilePath: dockerfileTemplate,
+		filepath.Join(metadataDir, "annotations.yaml"): annotationsTemplate,
+	}
+
+	for path, tmpl := range templateMap {
+		log.Info(fmt.Sprintf("Creating %s", path))
+		f, err := os.OpenFile(path, os.O_WRONLY|os.O_CREATE|os.O_TRUNC, 0666)
+		if err != nil {
+			return err
+		}
+
+		defer func() {
+			if err := f.Close(); err != nil {
+				log.Error(err)
+			}
+		}()
+		if err = tmpl.Execute(f, values); err != nil {
+			return err
+		}
+	}
+	log.Infof("Bundle metadata generated suceessfully")
+	return nil
+}
+
+// WriteScorecardConfig adds the default scorecard config containing OLM tests.
+func (meta *BundleMetaData) WriteScorecardConfig() error {
```

Comment:
The scorecard config should not be written to these bundles by this command. When the user chooses the latest bundle (ex. `mv ./bundles/bundle-1.2.3 ./bundle`) then runs `make bundle` for the first time, a scorecard config will be added to that bundle.

Reply:
A doubt.. why would the user choose to move the location of the latest bundle, given that we are providing the option to build the image which can be used with run bundle directly? In the sense, after converting pkgman to bundles why to run `make bundles`. If we don't write scorecard config here, then we would have to remove this too - [here](https://github.com/operator-framework/operator-sdk/blob/fa84c3a569e81f94d3c33fda17a4a7f30d31eac5/internal/util/bundleutil/template.go#L69). This would mean we would be building bundle images without scorecard config.

Reply:
These bundle images capture what the bundle would look like for some version within packagemanifests; if no scorecard config was present, then one should not be added by this command, nor should labels/annotations be present as you say. If there is a config, then these labels/annotations should be added.

Reply:
Have modified the implementation to look for scorecardconfig file and copy it to `tests/scorecard/config.yaml` if present.